index.js

```diff
@@ -1,88 +1,129 @@
-// console.log("%cQuestion 2", "color: red") 
+ console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
+ console.log(name)
 // => "Raffy"
 // name = "Not Raffy" 
 // => TypeError
-// console.log("%c----------", "color: red") 
+ console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
+
+function drinkWater(currentThirstLevel) {
+  console.log("Man I sure am thirsty");
+  currentThirstLevel -= 1;
+  console.log("Ahh that hits the spot");
+  return currentThirstLevel
+}
+
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+ console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
+
+function sameSameButDifferent (num, maybeNum) {
+  if (typeof num === "number" && num === maybeNum) {
+    return "same same"
+  } else if (num === parseInt(maybeNum, 10)) {
+    return "same same (but different)"
+  } else {
+    return "different"
+  }
+}
+
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
+
+function updateGrade(student, grade) {
+  student["grade"] = grade
+}
+
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%cQuestion 5", "color: red")
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)
+// => { name: "Duane", grade: 92 }
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
+
+function printNameAndPhones (users) {
+  for (let i = 0; i < users.length; i++) {
+    console.log(users[i]["name"]);
+    console.log(`Cell: ${users[i]["phones"]["cell"]}`);
+    console.log(`Office: ${users[i]["phones"]["office"]}`)
+  }
+}
+
```

Comment:
Good work! Javascript also has a couple other methods for iterating over arrays that make it a little easier:

- Here's how it would look with `for..of`:
```javascript
  for (let user of users) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  }
```

- Here's how it would look with `forEach`:
```javascript
  users.forEach(function(user) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  })